xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiServletURLFactory.java

```diff
@@ -444,23 +447,97 @@ public URL createExternalURL(String spaces, String name, String action, String q
         return this.createURL(spaces, name, action, querystring, anchor, xwikidb, context);
     }
 
+    private void appendQueryParameter(String key, Object paramValue, StringBuilder stringBuilder)
+        throws UnsupportedEncodingException
+    {
+        if (paramValue instanceof String) {
+            stringBuilder.append(URLEncoder.encode(key, "UTF-8"));
+            stringBuilder.append("=");
```

Comment:
```suggestion
            stringBuilder.append('=');
```

Better for performances.

Reply:
Right.